File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

float count_letters(string text)
{
    float n;

    n = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
            n++;
    }
    return (n);
}

float count_words(string text)
{
    float n;

    n = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
            n++;
    }
    return (n);
}

float count_sentences(string text)
{
    float n;

    n = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
            n++;
    }
    return (n);
}

float ft_readability(float letters, float words, float sentences)
{
    float L;
    float S;
    float index;

    L = (letters / words) * 100;
    S = (sentences / words) * 100;
    index = (0.0588 * L) - (0.296 * S) - 15.8;
    return (round(index));
}

int main(void)
{
    float letters;
    float words;
    float sentences;
    float readability;
    string text = get_string("Text: ");

    letters = count_letters(text);
    words = count_words(text);
    sentences = count_sentences(text);
    readability = ft_readability(letters, words, sentences);

    if (readability < 1)
        printf("Before Grade 1\n");
    else if (readability >= 1 && readability <= 16)
        printf("Grade %i\n", (int) readability);
    else if (readability > 16)
        printf("Grade 16+\n");
    return (0);
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0
for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    elif text[i].isspace():
        words += 1
    elif text[i] == '.' or text[i] == '!' or text[i] == '?':
        sentences += 1

l = letters / words * 100
s = sentences / words * 100
r = round(0.0588 * l - 0.29 * s - 15.8)
if r > 16:
    print("Grade 16+")
elif r > 1 and r < 17:
    print(f"Grade {r}")
else:
    print("Before Grade 1")
